va/va.c

```diff
@@ -673,14 +694,20 @@ VAStatus vaInitialize (
     va_MessagingInit();
 
     va_infoMessage(dpy, "VA-API version %s\n", VA_VERSION_S);
-
-    vaStatus = va_getDriverName(dpy, &driver_name);
-
+    /*get backend driver candidate number, by default the value should be 1*/
+    vaStatus = va_getDriverCandidateNum(dpy, &candidate_num);
+    if (vaStatus != VA_STATUS_SUCCESS) {
+        candidate_num = 1;
+    }
+    /*get driver name , get first driver candidate name, then try to load it*/
+    vaStatus = va_getDriverName(dpy, &driver_name, candidate_index);
```

Comment:
This code blows my mind... That's not about your changes, but about code which existed before... For example, can `ctx->override_driver_name` be changed in va_getDriverName() function? and if not - WHY we are calling `va_getDriverName` before it and not after?

Why we don't start process with analysis whether user specified LIBVA_DRIVER_NAME? I.e. from:
`if (driver_name_env && (geteuid() == getuid()))`

I kind of feel that this code can be rewritten to be much easier...

Reply:
@dvrogozh , I have to revert to previous change, because i965 driver need drm is authenticated, so va_drm_authenticate should be called anyway, and drm_state->auth_type = VA_DRM_AUTH_CUSTOM; should be set, if we refactor this part, it will not call this part with some conditions

Reply:
another solution is mv authenticate operation into vaGetNumCandidates function . I just update code.